### PR TITLE
Expand functionality of install script

### PR DESCRIPTION
**Pull request by jsf9k**
# 

## 🗣 Description

This pull request makes the following changes to the `00_install_freeipa.sh` script:
* Add ability to unenroll from the domain
* Add ability to enroll/re-enroll
* Default functionality is still to install

## 💭 Motivation and Context

When completely redeploying the FreeIPA cluster, it is sometimes desirable to unenroll clients from the domain and then re-enroll them afterwards.  It makes sense to add this functionality to the `00_install_freeipa.sh` script.

## 🧪 Testing

This functionality was successfully tested in our staging COOL environment.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: files/00_setup_freeipa.sh [hunk 1]
```diff
@@ -27,25 +27,79 @@ else
...
+}
+
+function unenroll {
+    ipa-join --unenroll
+    # hostname is defined in the FreeIPA variables file that is
+    # sourced toward the top of this file.  Hence we can ignore the
+    # "undefined variable" warning from shellcheck.
+    #
+    # shellcheck disable=SC2154
+    ipa-rmkeytab -p "host/$hostname" -k /etc/krb5.keytab
```
Comment (dav3r) [inline files/00_setup_freeipa.sh:76]: If `enroll` calls `ipa host-add-principal`, is there a reason why `unenroll` shouldn't call `ipa host-remove-principal`?

Comment (jsf9k) [reply files/00_setup_freeipa.sh:76]: Yes.  You may want to temporarily unenroll a host but _not_ remove it from the domain...the very reason why we decided to add this foo.

Comment (dav3r) [reply files/00_setup_freeipa.sh:76]: Got it- just making sure.  Thanks!

**Review by dav3r — APPROVED** [review]
This looks great to me! 👑

**Review by hillaryj — APPROVED** [review]
💥

**Pull request merged**